src/structs/Api.ts

```diff
@@ -0,0 +1,253 @@
+import fetch, { Headers } from 'node-fetch'
+import ApiError from '../utils/ApiError'
+import qs from 'querystring'
+import { EventEmitter } from 'events'
+
+interface APIOptions {
+  /**
+   * Top.gg Token
+   */
+  token: any
+  /**
+   * Whether or not to autopost
+   * @default true
+   */
+  autoPost?: boolean
+}
+
+/**
+ * Top.gg API Client for Posting stats or Fetching data
+ * @link https://top.gg/api/docs
+ * @example
+ * const Topgg = require('@top-gg/sdk')
+ *
+ * const api = new Topgg.Api('Your top.gg token')
+ * ```
+ */
+export class Api extends EventEmitter {
+  private options: APIOptions
+  /**
+   * Create Top.gg API instance
+   * @param token Token or options
+   * @param client Supported library client 
+   */
+  constructor (token: string, options?: APIOptions) {
+    super()
+    this.options = {
+      token: token,
+      autoPost: options.autoPost ?? true
+    }
+  }
+
+  private async _request (method: string, path: string, body?: object): Promise<any> {
+    const headers = new Headers()
+    if (this.options.token) headers.set('Authorization', this.options.token)
+    if (method !== 'GET') headers.set('Content-Type', 'application/json')
+
+    let url = `https://top.gg/api/${path}`
+
+    // @ts-ignore querystring typings are messed
+    if (body && method === 'GET') url += `?${qs.stringify(body)}`
+
+    const response = await fetch(url, {
+      method,
+      headers,
+      body: body && method !== 'GET' ? JSON.stringify(body) : null
+    })
+
+    let responseBody 
+    if (response.headers.get('Content-Type')?.startsWith('application/json')) {
+      responseBody = await response.json()
+    } else {
+      responseBody = await response.text()
+    }
+
+    if (!response.ok) {
+      throw new ApiError(response.status, response.statusText, responseBody)
+    }
+
+    return responseBody
+  }
+
+  /**
+   * Post bot stats to Top.gg (Do not use if you supplied a client)
+   * @param serverCount Server count
+   * @param shardId Current shard ID
+   * @param shardCount Total shard count
+   * @example
+   * ```js
+   * await client.postStats(28199, null, 1)
```

Comment:
Can these parameters be wrapped inside an interface like this instead?
```js
await client.postStats({ serverCount: 28199, shardCount: 1 })
```
Passing `undefined` or `null` into positional arguments is an antipattern